config/env/env_types.go

```diff
@@ -0,0 +1,161 @@
+package env
+
+import (
+	"errors"
+	"os"
+	"regexp"
+	"strconv"
+)
+
+var TypeError = errors.New("type could not be converted")
+var EnvironmentError = errors.New("environment variable not found")
+
+var regex *regexp.Regexp
+
+func init() { regex = regexp.MustCompile(`\${[A-Z]+[A-Z1-9_]*}`) }
+
+func replaceEnvVar(in string) (string, error) {
+
+	// TODO(ear7h): FindAllString
+	varName := regex.FindString(in)
+	// no env var
+	if len(varName) == 0 {
+		return in, nil
+	}
+
+	// trim the leading '${' and trailing '}'
+	varName = varName[2:len(varName)-1]
+
+	// get env var
+	envVar := os.Getenv(varName)
+	if envVar == "" {
+		return "", EnvironmentError
+	}
+
+	return regex.ReplaceAllString(in, envVar), nil
+}
+
+type Bool bool
+func BoolPtr(v Bool) *Bool {
+	return &v
+}
+
+func (t *Bool) UnmarshalTOML(v interface{}) error {
+	var boolVal bool
+	var err error
+
+	switch val := v.(type) {
+	case string:
+		val, err = replaceEnvVar(val)
+		if err != nil {
+			break
```

Comment:
I understand that you're catching the error farther down and returning, but I find this approach a bit confusing to follow. I think if we hav an error we should return then. I know it's extra code but it keeps things consistent.